chain/network/src/peer.rs

```diff
@@ -731,6 +743,20 @@ impl StreamHandler<Result<Vec<u8>, ReasonForBan>> for Peer {
                 return;
             }
         };
+        // Drop duplicated messages routed within 1s
+        if let PeerMessage::Routed(msg) = &peer_msg {
+            let key = (msg.author.clone(), msg.target.clone(), msg.signature.clone());
+            let now = Utc::now();
+            if let Some(time) = self.routed_message_cache.cache_get(&key) {
+                if time.signed_duration_since(now).num_milliseconds() as i128
+                    <= DROP_DUPLICATED_MESSAGES_PERIOD.as_millis() as i128
+                {
+                    debug!(target: "network", "Dropping duplicated message from {} to {:?}", msg.author, msg.target);
+                    return;
+                }
+            }
+            self.routed_message_cache.cache_set(key, now);
```

Comment:
alternative:

```rust
            if let Some(prev) = self.routed_message_cache.cache_set(key, now) {
                if prev.signed_duration_since(now).num_milliseconds() as i128
                    <= DROP_DUPLICATED_MESSAGES_PERIOD.as_millis() as i128
                {
                    debug!(target: "network", "Dropping duplicated message from {} to {:?}", msg.author, msg.target);
                    return;
                }
            }
```

that way we always refresh the timestamp of the message. Not sure if that's better or worse. If it is worse (ie, we do want to send duped messages eventually), it probably is worth a comment.

Reply:
@matklad Consider a situation when peer `A` sends a message to `B` every 10 ms. 
In my current implementation, we would drop 9 out 10 messages, and send one message every 100-100ms.

In the `alternative` you proposed, we would update the counter every 10ms and the message would never get send for the 2nd time. I think the original approach is more intuitive. @bowenwang1996 what do you think?